src/components/ProductList/ProductList.jsx

```diff
@@ -1,88 +1,109 @@
-            sortedProducts = sortedProducts.sort((a,b) => 
-            sort === "LOW_TO_HIGH" ? a.price - b.price : b.price - a.price 
+            sortedProducts = sortedProducts.sort((a, b) =>
+                sort === "LOW_TO_HIGH" ? a.price - b.price : b.price - a.price
             )
         }
-        if (byStock){
+        if (byStock) {
             sortedProducts = sortedProducts.filter((prod) => prod.stock);
         }
-        if (byFastDelivery){
+        if (byFastDelivery) {
             sortedProducts = sortedProducts.filter((prod) => prod.delivery);
         }
-        if (byNikon){
-            sortedProducts = sortedProducts.filter((prod) => prod.brand === byNikon || prod.brand === bySony || prod.brand === byCanon || prod.brand === bySamsung || prod.brand === byPanasonic );
+        if (byNikon) {
+            sortedProducts = sortedProducts.filter((prod) => prod.brand === byNikon || prod.brand === bySony || prod.brand === byCanon || prod.brand === bySamsung || prod.brand === byPanasonic);
         }
-        if (bySony){
+        if (bySony) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand === bySony || prod.brand === byNikon || prod.brand === byCanon || prod.brand === bySamsung || prod.brand === byPanasonic);
         }
-        if (byCanon){
+        if (byCanon) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand === byCanon || prod.brand === bySony || prod.brand === byNikon || prod.brand === bySamsung || prod.brand === byPanasonic);
         }
-        if (bySamsung){
+        if (bySamsung) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand === bySamsung || prod.brand === bySony || prod.brand === byCanon || prod.brand === byNikon || prod.brand === byPanasonic);
         }
-        if (byPanasonic){
+        if (byPanasonic) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand === byPanasonic || prod.brand === bySony || prod.brand === byCanon || prod.brand === bySamsung || prod.brand === byNikon);
         }
-        if (byHomeCategory){
+        if (byHomeCategory) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand === byHomeCategory)
         }
-        if (byRating){
+        if (byRating) {
             sortedProducts = sortedProducts.filter((prod) => prod.rating >= byRating)
         }
-        if (bySearch){
+        if (bySearch) {
             sortedProducts = sortedProducts.filter((prod) => prod.brand.toLowerCase().includes(bySearch))
         }
-        if (byRange){
-            sortedProducts = sortedProducts.filter((prod) => prod.price >= byRange )
+        if (byRange) {
+            sortedProducts = sortedProducts.filter((prod) => prod.price >= byRange)
         }
-         return sortedProducts
+        if (byCategory[0] !== undefined) {
+            sortedProducts = sortedProducts.filter((prod)=>byCategory.includes(prod.brand));
+        }
+        return sortedProducts
```

Comment:
all of this should be broken down to small functions for better code readability